private/bufpkg/bufmodule/bufmoduleref/bufmoduleref.go

```diff
@@ -416,10 +416,10 @@ func DependencyModulePinsForBucket(
 			dep.Remote,
 			dep.Owner,
 			dep.Repository,
-			dep.Branch,
+			"",
```

Comment:
Does the `NewModulePin` constructor still need to expose these parameters for other call-sites? Is this because we still have these fields represented in the Protobuf representation of `ModulePin`? Will they be removed later, or kept around forever?

Reply:
Yes, it's because they are still being represented in the Proto definition. And yeah, I think we will remove them when we are confident that all clients are upgraded (which is reasonable, since we'll require everyone to move to `v1` when we cut over the digest, since old clients cannot read new commits that have been pushed with the `b3` digest). So as a final clean-up after all the upgrades, I plan to cull down the Proto representation and handle it for both client and server.

Reply:
Make sense, especially because the constructor is used on the server-side.

We use the `CreateTime` attached to the `ModulePin` in our dependency resolution, so we'll need to make sure that's still possible with these changes.